src/Controls/tests/Core.UnitTests/PageTests.cs

```diff
@@ -579,5 +579,27 @@ public void LogicalChildrenDontAddToPagesInternalChildren()
 			Assert.Contains(customControl, page.LogicalChildrenInternal);
 			Assert.Contains(customControl, ((IVisualTreeElement)page).GetVisualChildren());
 		}
+
+		[Fact]
+		public void SendDisappearingUnsubscribesAppearingEvent()
```

Comment:
I think the new test fails:

![image](https://github.com/user-attachments/assets/ed9c781d-d70b-4d1c-8810-7db5570645c0)

Reply:
It was failing because the counter was not being reset, but I discovered it was not really reproducing the issue correctly so I will look into that today